chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -439,12 +439,8 @@ where
                             ConnectTransactionError::DelegationDataNotFound(delegation_id),
                         )?;
                     if !accounting_view.pool_exists(*delegation_data.source_pool())? {
-                        // clear the nonce
-                        self.account_nonce.insert(
-                            AccountType::Delegation(delegation_id),
-                            CachedOperation::Erase,
-                        );
-
+                        // When deleting a delegation nonce value is preserved in the db
+                        // in case reorg happens so we don't have to restore it
                         Some(
```

Comment:
Please make sure we're not doing anything else like this. We should only delete such information in a pruned database after the max reorg depth.

Reply:
> Please make sure we're not doing anything else like this.

I'd merge this and then I'll go thought the code and see what tests I can add.

> We should only delete such information in a pruned database after the max reorg depth.

we don't do this yet